client/src/NavBar.jsx

```diff
@@ -1,111 +1,121 @@
-  MDX: "mdx",
-  MISP: "misp",
-  MSDAX: "msdax",
-  MySQL: "mysql",
-  "Objective-C": "objective-c",
-  Pascal: "pascal",
-  Pascaligo: "pascaligo",
-  Perl: "perl",
-  Postiats: "postiats",
-  "Power Query": "powerquery",
-  PowerShell: "powershell",
-  "Protocol Buffers": "proto",
-  Python: "python",
-  "Q#": "qsharp",
-  R: "r",
-  Razor: "razor",
-  Redis: "redis",
-  Redshift: "redshift",
-  reStructuredText: "restructuredtext",
-  Ruby: "ruby",
-  Rust: "rust",
-  SB: "sb",
-  Scala: "scala",
-  Scheme: "scheme",
-  SCSS: "scss",
-  Shell: "shell",
-  Solidity: "sol",
-  AES: "aes",
-  SparkQL: "sparkql",
-  SQL: "sql",
-  ST: "st",
-  Swift: "swift",
-  SystemVerilog: "systemverilog",
-  Verilog: "verilog",
-  Tcl: "tcl",
-  Twig: "twig",
-  TypeScript: "typescript",
-  VB: "vb",
-  "WebGPU Shader Language": "wgsl",
-  XML: "xml",
-  YAML: "yaml",
-  JSON: "json",
-  PlainText: "plaintext",
-};
 
 export default function NavBar(props) {
   const [selectedLang, setSelectLang] = useState("PlainText");
+  const [search, setSearch] = useState("");
+
+  //console.log(value);
   const Location = useLocation();
   const { loginWithRedirect } = useAuth0();
   const { logout } = useAuth0();
   const { user, isAuthenticated } = useAuth0();
   useEffect(() => {
-    if (location.pathname.split("/")[2]) {
+    /*if (location.pathname.split("/")[2]) {
       axios
-        .get(`localhost:9001/${location.pathname.split("/")[2]}`)
+        .get(
+          `https://codesharebackendapi.onrender.com/${
+            location.pathname.split("/")[2]
+          }`
+        )
         .then((res) => {
           const selectedKey = Object.keys(supportedLangList).find(
             (key) => res.data.language == supportedLangList[key]
-          );
-          setSelectLang(selectedKey);
-        });
-    }
+          );*/
+    const selectedKey = "PlainText";
+    const lastSelectedLanguage = getLanguage(LAST_SELECTED_LANGUAGE);
+    console.log(lastSelectedLanguage);
+    setSelectLang(lastSelectedLanguage ? lastSelectedLanguage : selectedKey);
+    //});
+    //}
   }, []);
-
-  const onSelectLang = (value) => {
+  function onSelectLang(value) {
     setSelectLang(value);
+    setLanguage(LAST_SELECTED_LANGUAGE, value);
+    const existingLanguage = JSON.parse(getLanguage(LANGUAGE_HISTORY)) || [];
+    existingLanguage.includes(value)
+      ? existingLanguage.splice(existingLanguage.indexOf(value), 1)
+      : existingLanguage;
+    existingLanguage.push(value);
+    setLanguage(LANGUAGE_HISTORY, JSON.stringify(existingLanguage));
     props.setLang(() => supportedLangList[value]);
+  }
+
+  const filteredLanguagesList = useMemo(() => {
+    const results = [];
+
+    for (let key in supportedLangList) {
+      const currentLang = supportedLangList[key];
+
+      if (
+        search &&
+        currentLang.toLowerCase().search(search.toLowerCase()) !== -1
+      ) {
+        results.push({ id: currentLang, value: key });
+      } else if (!search) {
```

Comment:
@gunjanbajaj1 - Convert this into a guard-block code

Reply:
@xDAnkit Fixed